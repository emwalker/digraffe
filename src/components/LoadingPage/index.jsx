// @flow
import React from 'react'
import Octicon from 'react-component-octicons'

type Props = {
  location: {
    pathname: string,
    state: ?Object,
  }
}

const LoadingPage = ({ location }: Props) => {
  const state = location.state || {}
  const { orgLogin, repoName, itemTitle } = state

  return (
    <div>
      <nav aria-label="Breadcrumb" className="mb-1">
        <ol>
          <li className="breadcrumb-item">
            <Octicon name="repo" className="mr-1" />
            {' '}
            { orgLogin }
          </li>
          <li
            className="breadcrumb-item breadcrumb-item-selected text-gray"
            aria-current="page"
          >
            { repoName }
          </li>
        </ol>
      </nav>
      <div className="Subhead">
        <div className="Subhead-heading">
          { itemTitle }
        </div>
      </div>
    </div>
  )
}

export default LoadingPage
