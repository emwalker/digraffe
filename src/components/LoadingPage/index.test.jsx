import React from 'react'
import { shallow } from 'enzyme'

import LoadingPage from './index'

const props = {
  location: {
    pathname: '/some/path',
    state: {
      orgLogin: 'Gnusto',
      repoName: 'General collection',
      itemTitle: 'Some title',
    },
  },
}

describe('<LoadingPage />', () => {
  const wrapper = shallow(<LoadingPage {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
