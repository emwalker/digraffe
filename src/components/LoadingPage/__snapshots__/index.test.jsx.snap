// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingPage /> renders 1`] = `
<Page>
  <nav
    aria-label="Breadcrumb"
    className="mb-1"
  >
    <ol>
      <li
        className="breadcrumb-item"
      >
        <GoRepo
          className="mr-1"
        />
         
        <a
          href="#"
        >
          Gnusto
        </a>
      </li>
      <li
        aria-current="page"
        className="breadcrumb-item breadcrumb-item-selected text-gray"
      >
        General collection
      </li>
    </ol>
  </nav>
  <div
    className="Subhead clearfix gutter"
  >
    <div
      className="Subhead-heading col-lg-8 col-12"
    >
      Some title
    </div>
    <SearchBox
      className="col-lg-4 col-12"
      onEnter={null}
      value=""
    />
  </div>
  <div>
    <Columns>
      <RightColumn>
        <_default
          items={Array []}
          orgLogin="Gnusto"
          placeholder="There are no parent topics for this topic."
          repoName="General collection"
          title="Parent topics"
        />
      </RightColumn>
      <LeftColumn>
        <div
          className="blankslate"
        >
          <p>
            Searching the computers for items ...
          </p>
        </div>
      </LeftColumn>
    </Columns>
  </div>
</Page>
`;
